### PR TITLE
Replaced default with individual states

### PR DESCRIPTION
**Pull request by amadrafi**
I replaced all the defaults statements and for brakes, I wasn't sure about the critical failure function so I put navigation as a placeholder for kInvalid.

### CODE REVIEW

File: src/sensors/gpio_manager.cpp [hunk 1]
```diff
@@ -94,7 +94,9 @@ void GpioManager::run()
           setHP();
           log_.INFO("GPIO-MANAGER", "kReady...HP SSR set and HP on");
           break;
-        default:      // undefied behaviour, e.g. kInvalid
+
+        //Default for all the other states
+        case data::State::kInvalid:      // undefied behaviour, e.g. kInvalid
```
Comment (MiltFra) [inline src/sensors/gpio_manager.cpp:99]: Remove the `e.g. kInvalid`

File: src/propulsion/main.cpp [hunk 2]
```diff
@@ -73,7 +73,11 @@ void Main::run()
       case data::State::kEmergencyBraking:
         state_processor_->quickStopAll();
         break;
-      default:
+
```
Comment (MiltFra) [inline src/propulsion/main.cpp:76]: Remove the empty line

File: src/navigation/main.cpp [hunk 3]
```diff
@@ -55,9 +55,14 @@ void Main::run()
         nav_.navigate();
         break;
 
-      default:
+      //Default for all the other states
+      case State::kFailureStopped:
+      case State::kFinished:
         navigation_complete = true;
         break;
+
```
Comment (MiltFra) [inline src/navigation/main.cpp:63]: empty line

File: src/navigation/main.cpp [hunk 4]
```diff
@@ -55,9 +55,14 @@ void Main::run()
         nav_.navigate();
         break;
 
-      default:
+      //Default for all the other states
```
Comment (MiltFra) [inline src/navigation/main.cpp:58]: i don't think we need this comment; it only makes sense in this context, i.e. it won't give any insight in the future

File: src/brakes/main.cpp [hunk 5]
```diff
@@ -115,7 +115,15 @@ void Main::run()
           f_brake_->checkAccFailure();
         }
         break;
-      default:
+
+      //Default for all the other states
```
Comment (MiltFra) [inline src/brakes/main.cpp:119]: ...

**Review by MiltFra — CHANGES_REQUESTED** [review]
Looking at the warnings generated by the compiler, I've also realised that we need to move `num_states` out of the enum into a constexpr `kNumStates`. This will require you to touch quite a few files.

File: src/brakes/main.cpp [hunk 6]
```diff
@@ -115,7 +115,12 @@ void Main::run()
           f_brake_->checkAccFailure();
         }
         break;
-      default:
+      case data::State::kReady:
+      case data::State::kFailureStopped:
+        break;
+      case data::State::kInvalid:
+        brakes_               = data_.getEmergencyBrakesData();
+        brakes_.module_status = ModuleStatus::kCriticalFailure;
```
Comment (MiltFra) [inline src/brakes/main.cpp:123]: You need to write back to data to set critical failure. Also, I'd assume we want to handle this similarly to `kEmergencyBraking`.

File: src/navigation/main.cpp [hunk 7]
```diff
@@ -54,10 +54,12 @@ void Main::run()
...
         nav_.navigate();
         break;
-
-      default:
+      case State::kFailureStopped:
+      case State::kFinished:
         navigation_complete = true;
         break;
+      case State::kInvalid:
+        break;
```
Comment (MiltFra) [inline src/navigation/main.cpp:62]: We probably don't want to do nothing here. Either we stop navigation to free up processing to make sure we can shut down properly or we keep navigating to keep reporting displacement/velocity/acceleration.

**Review by MiltFra — CHANGES_REQUESTED** [review]

**Review by SnickeyX — APPROVED** [review]

**Review by MiltFra — APPROVED** [review]

**Pull request merged**